lib/daru/view/adapters/googlecharts.rb

```diff
@@ -14,24 +14,32 @@ module GooglechartsAdapter
         # the google charts option concept.
         # and google_visualr : http://googlevisualr.herokuapp.com/
         #
-        # TODO : this docs must be improved
+        # @param data [Array, Daru::DataFrame, Daru::Vector, Daru::View::Table,
+        #   String] The data provided by the user to generate the google chart.
+        #   Data in String format represents the URL of the google spreadsheet
+        #   from which data has to invoked
+        # @param options [Hash] Various options provided by the user to
+        #   incorporate in google charts
+        # @return [GoogleVisualr::Interactive] Returns the chart object based
+        #   on the chart_type
         def init(data=[], options={})
           @table = GoogleVisualr::DataTable.new
-          @table =
-            if data.is_a?(Daru::View::Table) && data.table.is_a?(GoogleVisualr::DataTable)
-              data.table
-            elsif data.is_a?(GoogleVisualr::DataTable)
-              data
-            else
-              add_data_in_table(data)
-            end
+          @table = get_table(data) unless data.is_a?(String)
```

Comment:
What will happen if string is not a URL?

Reply:
What will be `plot.table` when we have URL?

Reply:
plot.table will contain the empty table as the DataTable is generated in query response [here](https://github.com/SciRuby/daru-view/pull/88/files#diff-873104e004ae7bc85b74e657779a5804R50) and we can not retrieve the data from google spreadsheet.

Reply:
We might be able to retrieve the data from the URL provided. It is because the URL provides the access to JSON data which can then be stored in a Daru::DataFrame.

Reply:
We can get the table , [as here](https://github.com/SciRuby/daru-view/pull/88/files#diff-cd0c826056999c96e0390d69150cc07bR107).